server/webapp/WEB-INF/rails.new/app/assets/new_javascripts/views/agents/agent_table_header.js.msx

```diff
@@ -17,31 +17,110 @@
 define(['mithril'], function (m) {
 
   var AgentsTableHeader = {
+
+    controller: function (args) {
+
+      var sortType = function () {
+        return m.route.param('sortBy');
+      };
+
+      var orderType = function () {
+        return m.route.param('orderBy')
+      };
+
+      return {
+        sortOrder: function (sortBy) {
+          var attrName = sortType() || 'agentState';
+          var orderBy  = orderType() || 'asc';
+          return _.isEqual(attrName, sortBy) ? orderBy : '';
+        },
+
+        toggleSortingOrder: function (sortBy) {
+          var sortingOrder = _.isEqual(sortBy, sortType()) ? (_.isEqual(orderType(), 'asc') ? 'desc' : 'asc') : 'asc';
+          args.sortBy(sortBy, sortingOrder);
+        }
+      }
+    },
+
     view: function (ctrl, args) {
       return (
-        <thead>
+        <thead style={args.style}>
         <tr>
-          <td width="2%">
+          <th>
             <input type="checkbox"
                    checked={args.checkboxValue()}
-                   onclick={m.withAttr('checked', function(value) {
-                      args.onCheckboxClick(value);
-                   })}
+                   onclick={m.withAttr('checked', args.onCheckboxClick)}
             />
-          </td>
-          <td width="13%">Agent Name</td>
-          <td width="5%">Sandbox</td>
-          <td width="12%">OS</td>
-          <td width="10%">IP Address</td>
-          <td width="5%">Status</td>
-          <td width="8%">Free Space</td>
-          <td width="25%">Resources</td>
-          <td width="20%">Environments</td>
+          </th>
+          <th>
+            <label class="agents-table-header" onclick={ctrl.toggleSortingOrder.bind(ctrl, 'hostname')}>Agent Name
+              <div class={"sort " + ctrl.sortOrder('hostname')}>
+                <span class="up"></span>
+                <span class="down"></span>
+              </div>
+            </label>
+          </th>
+          <th>
+            <label class="agents-table-header" onclick={ctrl.toggleSortingOrder.bind(ctrl, 'sandbox')}>Sandbox
+              <div class={"sort " + ctrl.sortOrder('sandbox')}>
```

Comment:
Should it be possible to extract this div into some `<sortHeader/>` component? Does not need to be a public component in its own file, even something local will do.